app/api/chemotion/sample_api.rb

```diff
@@ -74,36 +74,19 @@ class SampleAPI < Grape::API
           # Create a temp file in the tmp folder and sdf delayed job, and pass it to sdf delayed job
           extname = File.extname(params[:file][:filename])
           if extname.match(/\.(sdf?|mol)/i)
-            sdf_import = Import::ImportSdf.new(file_path: params[:file][:tempfile].path,
```

Comment:
Suggestion: Extract the logic into  separate methods: 
line 76 - 90 -> importFromMolFile ( )
line 91 - 120 -> importFromXlsCsv ( ) -> here you could even split it again in immediatelyImport ( ) and jobBasedImport ( )